test/clients/azure/osa.go

```diff
@@ -24,3 +31,26 @@ func (cli *Client) OSAResourceGroup(ctx context.Context, resourcegroup, name, lo
 	rgid := *app.ApplicationProperties.ManagedResourceGroupID
 	return rgid[strings.LastIndexByte(rgid, '/')+1:], nil
 }
+
+// UpdateOSACluster updates an OpenshiftManagedCluster by providing both the
+// currently applied external manifest and an updated internal manifest
+// NOTE: This method will go away as soon as the admin API is ready. Until then,
+// only the KeyRotation or other e2e tests needing admin functionality should
+// call this method.
+func (cli *Client) UpdateOSACluster(ctx context.Context, external *v20180930preview.OpenShiftManagedCluster, config *api.PluginConfig) (*v20180930preview.OpenShiftManagedCluster, error) {
+	// Remove the provisioning state before updating, if set
+	external.Properties.ProvisioningState = nil
+
+	// create logger
+	logrus.SetLevel(log.SanitizeLogLevel("Debug"))
+	logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+	logger := logrus.NewEntry(logrus.StandardLogger())
+
+	var oc *v20180930preview.OpenShiftManagedCluster
+	var err error
+	// simulate the API call to the RP
+	if oc, err = fakerp.CreateOrUpdate(ctx, external, logger, config); err != nil {
```

Comment:
Here you should be using the OSA client we have in https://github.com/openshift/openshift-azure/blob/master/pkg/api/2018-09-30-preview/api/client.go#L94, similar to how it's used in createorupdate instead of using the fake RP backend function.

Reply:
we'll take care in a follow up #807

Reply:
Actually I still think we should be fixing this here. After the admin API lands, you still need a client call from this test so `UpdateOSACluster` is not going to be removed but repurposed, so I think it's worth structuring it correctly from start, ie. you should be querying the fake RP instead of running the update yourself.

Reply:
Should be fixed https://github.com/openshift/openshift-azure/pull/795 where the OSA client is re-introduced into the e2e framework.